src/OpenTelemetry.Api/CHANGELOG.md

```diff
@@ -6,6 +6,9 @@ Notes](../../RELEASENOTES.md).
 
 ## Unreleased
 
+* Revert optimize performance of `TraceContextPropagator.Extract` from #5749.
```

Comment:
We can resolve https://github.com/open-telemetry/opentelemetry-dotnet/issues/6158 with this change, but need a way to make it visible in changelog. 

 I suggest adding https://github.com/open-telemetry/opentelemetry-dotnet/issues/6158 in this changelog, and also in the instrumentation library changelog too when they bump api version

Reply:
I've tweaked the changelog message.